src/main/java/randoop/main/GenInputsAbstract.java

```diff
@@ -1147,6 +1173,145 @@ private static boolean shouldOmitClass(String classname) {
     }
   }
 
+  /**
+   * Returns classes in the given package. Does not include classes in sub-packages. Ignores
+   * non-accessible classes.
+   *
+   * @param packageName a package name
+   * @param accessibility the accessibility predicate
+   * @return classes in package {@code packageName}
+   */
+  public static Set<@ClassGetName String> getClassnamesFromPackage(
+      String packageName, AccessibilityPredicate accessibility) {
+    Set<@ClassGetName String> classnames = new HashSet<>();
+
+    for (File location : getClassLocationsForCurrentClasspath()) {
+      if (location.isFile() && location.getName().endsWith(".jar")) {
+        classnames.addAll(searchInJar(location, packageName, accessibility));
+      } else if (location.isDirectory()) {
+        classnames.addAll(searchInDirectory(location, packageName, accessibility));
+      }
+    }
+
+    return classnames;
+  }
+
+  /**
+   * Searches in the given directory for classes with the given package.
+   *
+   * @param directory a directory
+   * @param packageName a package name
+   * @param accessibility the accessibility predicate
+   * @return classes with the given package that were found in the specified directory
+   */
+  public static Set<@ClassGetName String> searchInDirectory(
+      File directory, String packageName, AccessibilityPredicate accessibility) {
+    String packageNameAsFile = packageName.replace(".", File.separator) /*.concat(File.separator)*/;
+    File packageDirectory =
+        new File(directory.getPath().concat(File.separator).concat(packageNameAsFile));
+    if (packageDirectory.exists()) {
+      try (Stream<Path> allFiles = Files.walk(Paths.get(packageDirectory.getPath()))) {
+        @SuppressWarnings("signature")
+        Set<@ClassGetName String> classnames =
+            allFiles
+                .filter(e -> e.toFile().isFile()) // to ignore directories
+                .map(Path::toString)
+                .map(
+                    absolutePath ->
+                        absolutePath.replace(
+                            directory.getPath() + File.separator, "")) // to delete absolute path
+                .filter(
+                    relativePath -> relativePath.startsWith(packageNameAsFile)) // to filter package
+                .filter(relativePath -> relativePath.endsWith(".class")) // to only get classes
+                .map(
+                    relativePath ->
+                        relativePath.substring(0, relativePath.length() - ".class".length()))
+                .map(ifClassName -> ifClassName.replace(File.separator, "."))
+                .filter(
+                    classname -> {
+                      try {
+                        return classname.startsWith(packageName)
```

Comment:
It is awkward to do a partial test for the package name on line 1224 and more here.  Please put the entire test for package name in one place.

Reply:
All tests, except accessibility.isAccessible are now placed in filter provided to listFiles method. accessibility.isAccessible test is separate because it tests class, not a file or filename. Also, some additional operations to perform this check are needed, e. g. transformation of filename to ClassGetName form.